channels/ON-CHAIN.md

```diff
@@ -102,13 +120,19 @@ This operation is not mandatory for normal channel operation.
 Serialization defined [here](../serializations.md#channel-deposit-transaction)
 
 
-- `channel_id`: channel id as recorded on chain
+- `channel_id`: channel id as recorded on-chain
 - `from`: sender of the deposit
 - `amount`: amount of tokens deposited
 - `ttl`:
 - `fee`:
+- `state_hash`: the root hash of the channel state tree after the deposit had been applied
```

Comment:
Is `state_hash` in this `channel_deposit` tx distinct from the `state_hash` in the channel object in the block state a requirement for consensus? I would expect yes.

And in the very unlikely case of hash conflict it shall be resolved at higher level (e.g. by changing ttl field or similar trickery).